### ISSUE TITLE
ReflectionException: Class does not exist (since v1.14.0)

### ISSUE THREAD

**Issue by eliashaeussler** [CuyZ/Valinor#574]
Hi,

I experience the following issue since v1.14.0:

> ReflectionException: Class "EliasHaeussler\CacheWarmup\Sitemap" does not exist

I have the following class:

```php
namespace EliasHaeussler\CacheWarmup\Result;

use EliasHaeussler\CacheWarmup\Sitemap;

final class ParserResult
{
    /**
     * @param list<Sitemap\Sitemap> $sitemaps
     * @param list<Sitemap\Url>     $urls
     */
    public function __construct(
        private readonly array $sitemaps = [],
        private readonly array $urls = [],
    ) {}

    /**
     * @return list<Sitemap\Sitemap>
     */
    public function getSitemaps(): array
    {
        return $this->sitemaps;
    }

    /**
     * @return list<Sitemap\Url>
     */
    public function getUrls(): array
    {
        return $this->urls;
    }
}
```

(https://github.com/eliashaeussler/cache-warmup/blob/3.2.0/src/Result/ParserResult.php)

It seems there's an issue with the `Sitemap` class where `Sitemap` is also part of the namespace (`EliasHaeussler\CacheWarmup\Sitemap\Sitemap`) and only the short namespace (`EliasHaeussler\CacheWarmup\Sitemap`) is used as class import, whereas classes are referenced by the short namespace (`Sitemap\Sitemap`).

When using complete class imports, everything works fine:

```diff
diff --git a/src/Result/ParserResult.php b/src/Result/ParserResult.php
index a2664f2..42771fe 100644
--- a/src/Result/ParserResult.php
+++ b/src/Result/ParserResult.php
@@ -23,7 +23,8 @@ declare(strict_types=1);
 
 namespace EliasHaeussler\CacheWarmup\Result;
 
-use EliasHaeussler\CacheWarmup\Sitemap;
+use EliasHaeussler\CacheWarmup\Sitemap\Sitemap;
+use EliasHaeussler\CacheWarmup\Sitemap\Url;
 
 /**
  * ParserResult.
@@ -36,8 +37,8 @@ use EliasHaeussler\CacheWarmup\Sitemap;
 final class ParserResult
 {
     /**
-     * @param list<Sitemap\Sitemap> $sitemaps
-     * @param list<Sitemap\Url>     $urls
+     * @param list<Sitemap> $sitemaps
+     * @param list<Url>     $urls
      */
     public function __construct(
         private readonly array $sitemaps = [],
@@ -45,7 +46,7 @@ final class ParserResult
     ) {}
 
     /**
-     * @return list<Sitemap\Sitemap>
+     * @return list<Sitemap>
      */
     public function getSitemaps(): array
     {
@@ -53,7 +54,7 @@ final class ParserResult
     }
 
     /**
-     * @return list<Sitemap\Url>
+     * @return list<Url>
      */
     public function getUrls(): array
     {
```

* Related PR: https://github.com/eliashaeussler/cache-warmup/pull/408
* Related test run: https://github.com/eliashaeussler/cache-warmup/actions/runs/11666093210/job/32480260612?pr=408

**Comment by romm** [comment]
Hi @eliashaeussler thanks for the detailed report, sorry for the inconvenience.

This should be fixed with #575, could you try it on your side see if it solved the issue?

When we're good to go, I'll publish a new release so that you can enjoy the new versions. 😉

**Comment by eliashaeussler** [comment]
Hi @romm, thanks for the quick response and fix! 😊 It indeed resolves my issue, all tests are green again 💪 PR is approved :) Thanks again and have a nice day!

**Comment by romm** [comment]
Released in [`1.14.1`](https://github.com/CuyZ/Valinor/releases/tag/1.14.1), enjoy!

**Comment by eliashaeussler** [comment]
Thank you! ❤️
